### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by davidbanham**
Add basic request logging with morgan

### CODE REVIEW

File: base/package.mustache [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "coverage": "istanbul cover _mocha",
     "start": "node src/server",
     "test": "NODE_ENV=test mocha",
-    "watch": "mocha --watch",
+    "watch": "NODE_ENV=test mocha --watch",
```
Comment (MadRabbit) [inline base/package.mustache]: probably should ditch the whole `"watch"` thing in favour of running `npm test -- --watch`

**Comment by MadRabbit** [comment]
i reckon the consensus was that the team prefers `banyan` instead of `morgan`, no?

**Comment by MadRabbit** [comment]
also there is the `src/utils/logger`, i reckon the access log should use that instead of a separate logs engine

File: base/package.mustache (hunk 1 above)
Comment (davidbanham) [inline base/package.mustache]: Agreed

**Comment by davidbanham** [comment]
I thought that chat was about Bunyan vs Winston. I'm not tied to Morgan at all. It was just the one recommended by express in the 3 to 4 migration guide.

Probably right on the utils logger. I'll take a look at it.

**Comment by davidbanham** [comment]
Morgan is dead. Logs go through util instead.

Would be more betterer to have the option to run the access logger after the routes have been run (or both places) so that we can log the response code too. At the moment we can't do that because none of our routes call `next()`

Originally that was just an oversight on my part. Should we just sort it so that routes call next() when they're done or is there a specific reason for not doing so?

**Comment by MadRabbit** [comment]
weird, the status code works just fine with morgan for me, without calling `next()`. i'm not sure if you have to call it. i guess we need to investigate this problem, maybe make `express-yields` to call it or something. not sure

**Comment by davidbanham** [comment]
Morgan runs early in the stack and then defers the log event internally. We could do the same, I suppose. Just seemed simpler to mount the logger after the routes in the stack. I suppose it might be more error prone if someone forgets to call next, though.

**Comment by MadRabbit** [comment]
yeah, i'd use some already existing banyan <-> express plugins as well

**Comment by davidbanham** [comment]
The only one I saw was unmaintained. You know a good one?

**Comment by MadRabbit** [comment]
i used this once and it kind of worked https://github.com/villadora/express-bunyan-logger
there is also your friend's Eric Elliot's package https://github.com/ericelliott/bunyan-request-logger

**Comment by davidbanham** [comment]
The villadora one is the unmaintained one I saw. I'll check out the Eric one.

**Comment by davidbanham** [comment]
Hmm. Eric's thing just logs out the entire req and res objects. I don't think we want to be logging POST bodies as a default. It also instantiates its own copy of Bunyan, so we'd either need to monkey patch it or something in the middleware or require it instead of Bunyan in `utils/logger`.

While in there I saw that the only magic is waiting for the `finish` event on req, so I've dropped that in.

**Pull request merged**